File: Chain of Responsibility/Dart/Authentication.dart
import 'Handler.dart';
import 'Response.dart';
import 'Request.dart';

class Authentication extends Handler {
  @override
  Response? execute(Request? request) {
    if (request?.is_authenticated == true) {
      print("Authenticated");

      return next_handler?.execute(request);
    } else {
      print("Unauthenticated");

      return Response(
        403, 
        'Unauthenticated', 
        null
      );
    }
  }
}

File: Chain of Responsibility/Javascript/Authentication.js
import Handler from "./Handler.js";
import Response from "./Response.js";

class Authentication extends Handler {
  execute(request) {
    if (request.authenticated === true) {
      console.log("Authenticated");

      return this.next_handler.execute(request);
    } else {
      return new Response(403, "Unauthenticated", null);
    }
  }
}

export default Authentication;